### DISCUSSION TITLE
ESP32-A2DP with PlatformIO

### DISCUSSION THREAD

**Discussion by rbagdazian** [pschatzmann/ESP32-A2DP#368] [Q&A] [answered]
I wanted to switch over to the PlatformIO extension to VSCode for building the ESP32-A2DP project. Everything went well but I found that I had to add #include <Arduino.h> to the src/main.cpp file prior to the #include "BluetoothA2DPSink.h" or else I received an error that said:
```
src/main.cpp: In function 'void setup()':
src/main.cpp:7:3: error: 'Serial' was not declared in this scope
   Serial.begin(115200);
   ^~~~~~
src/main.cpp:7:3: note: suggested alternative: 'erfl'
   Serial.begin(115200);
   ^~~~~~
   erfl
```
I received this error for both of the platform.ini examples you provided in the example code. I was wondering if this is expected or did I miss a step somewhere. Thank you.

**Comment by pschatzmann** [comment, marked as answer]
If I remember right this is a feature of PlatformIO and you need to add #include <Arduino.h> explicitly in your sketch if you want to use functionality from Arduino.
You might get away if you include an other Arduino library which already has this include.  ESP32-A2DP is definitely not one of them because it does not depend on Arduino!

I have updated the example...

**Comment by rbagdazian** [comment]
Ok, got it. Looks good. Thanks for your kind assistance!
